### ISSUE TITLE
equivalent to pythons FileDescriptorProto.parseFromString() ?

### ISSUE THREAD

**Issue by tograd** [dcodeIO/protobuf.js#1077]
https://github.com/sysdream/Protod/blob/b683a9cad4ed32e4e1d9e67e8888fb9da73dca0f/protod#L276-L277

trying to rewrite protod in node but can't figure out how to do this with protobufjs 

what functions and what imports? tips appreciated

**Comment by dcodeIO** [comment]
Seems that `parseFromString` decodes a protobuf message from a binary-encoded string. With protobuf.js, that's what `SomeMessage.decode` does, but it doesn't take a string but an `Uint8Array` as the input.

**Comment by tograd** [comment]
Thank you!

Got stuck for a long while before I found out binary files should be encoded in latin1... 

Anyway, I've come a long way now (99% done except for refactoring) but I've got a weird issue yet. Most of my binary metadata is decoded correctly, except for one of them for some reason. The decode function doesn't properly include the package name..

Can't really reason as to what exactly is the issue. I've tried the following:

1. Use the binary descriptor files that python generates in the node app (same issue, it ignores package name for example)
2. Use the binary descriptor files that node generates in the python app (generates `DecodeError: Unexpected end-group tag.`)

I've included the binary file (.protoc), the decoded file(json.stringify) and the finalized .proto file. `monsters...` is the one I got issue with, `appearances...` is perfect and I can't see the relevant difference in their binary data. `monsters` seem to have a lot of extra data at the end but the protobufjs decode function doesn't seem to care (python version rejects it though as noted above)

[node.zip](https://github.com/dcodeIO/protobuf.js/files/2167537/node.zip)

I decode with `protobufjs\google\protobuf\descriptor.proto`

Say if you want something more from me. Thanks.
